sympy/core/numbers.py

```diff
@@ -2914,13 +2936,17 @@ def __sub__(self, other):
                 return S.NaN
             elif other.is_Float:
                 if other == Float('-inf'):
-                    return Float('nan')
+                    return S.NaN
                 else:
                     return Float('-inf')
             else:
                 return S.NegativeInfinity
         return NotImplemented
 
+    @_sympifyit('other', NotImplemented)
+    def __rsub__(self, other):
+        return (-self).__sub__(other)
```

Comment:
?

Reply:
Yes, this should be `__add__`. The purpose is to keep `1.0 - +/-oo` from becoming `-float(+/-oo) + 1.0`.